### DISCUSSION TITLE
Recording for this session is not available.

### DISCUSSION THREAD

**Discussion by PallaviPalisetti** [gravitational/teleport#20105] [Q&A]
Hi there,
I have installed teleport opensource version v11.2.0. there was an issue with session recordings of node.

DEBU [SESSION:N] Copying from PTY to writer completed with error <nil>. session_id:89d3e83b-3646-4a42-875b-2d33d83bd34e srv/sess.go:926
DEBU [SESSION:N] Copying from reader to PTY completed with error <nil>. session_id:89d3e83b-3646-4a42-875b-2d33d83bd34e srv/sess.go:934
ERRO [SESSION:N] Timed out waiting for PTY copy to finish, session data  may be missing. session_id:9a22a311-5d41-427f-8019-d14bf29941fb

DEBU [SESSION:N] Copying from PTY to writer completed with error read /dev/ptmx: input/output error. session_id:9a22a311-5d41-427f-8019-
.go:926
DEBU [SESSION:N] Copying from reader to PTY completed with error <nil>. session_id:9a22a311-5d41-427f-8019-d14bf29941fb srv/sess.go:934
DEBU [TERM:LOCA] Closed PTY srv/term.go:310

**Comment by webvictim** [comment]
Is the session recording actually missing? That error can sometimes be benign.

**Reply by PallaviPalisetti** [reply]
Yes, missing 
 DEBU [AUDIT]     GetSessionEvents. afterN:0 printEvents:true sid:89d3e83b-3646-4a42-875b-2d33d83bd34e events/auditlog.go:760
2023-01-11T19:24:00Z DEBU [AUDIT]     Starting download of 89d3e83b-3646-4a42-875b-2d33d83bd34e. events/auditlog.go:533
2023-01-11T19:24:00Z DEBU [WEB]       Unable to find events for session 89d3e83b-3646-4a42-875b-2d33d83bd34e. error:[
ERROR REPORT:
Original Error: *trace.BadParameterError AccessDenied: Access Denied
        status code: 403, request id: 92d8b8b3-0a8d-4153-8fa3-8b076db19f22, host id:
Stack Trace:

Caught:
        github.com/gravitational/teleport/lib/httplib/httplib.go:182 github.com/gravitational/teleport/lib/httplib.ConvertResponse
        github.com/gravitational/teleport/lib/auth/clt.go:269 github.com/gravitational/teleport/lib/auth.(*Client).Get
        github.com/gravitational/teleport/lib/auth/clt.go:1109 github.com/gravitational/teleport/lib/auth.(*Client).GetSessionEvents
        github.com/gravitational/teleport/lib/web/apiserver.go:3079 github.com/gravitational/teleport/lib/web.(*Handler).siteSessionEventsGet
        github.com/gravitational/teleport/lib/web/apiserver.go:3234 github.com/gravitational/teleport/lib/web.(*Handler).WithClusterAuth.func1
        github.com/gravitational/teleport/lib/httplib/httplib.go:95 github.com/gravitational/teleport/lib/httplib.MakeHandlerWithErrorWriter.func1
        github.com/julienschmidt/httprouter@v1.3.0/router.go:399 github.com/julienschmidt/httprouter.(*Router).ServeHTTP
        net/http/server.go:2152 net/http.StripPrefix.func1
        net/http/server.go:2109 net/http.HandlerFunc.ServeHTTP
        github.com/gravitational/teleport/lib/web/apiserver.go:364 github.com/gravitational/teleport/lib/web.NewHandler.func1
        net/http/server.go:2109 net/http.HandlerFunc.ServeHTTP
        github.com/julienschmidt/httprouter@v1.3.0/router.go:460 github.com/julienschmidt/httprouter.(*Router).ServeHTTP
        github.com/gravitational/teleport/lib/web/apiserver.go:259 github.com/gravitational/teleport/lib/web.(*APIHandler).ServeHTTP
        github.com/gravitational/oxy@v0.0.0-20221029012416-9fbf4c444680/ratelimit/tokenlimiter.go:118 github.com/gravitational/oxy/ratelimit.(*TokenLimiter).ServeHTTP
        github.com/gravitational/oxy@v0.0.0-20221029012416-9fbf4c444680/connlimit/connlimit.go:75 github.com/gravitational/oxy/connlimit.(*ConnLimiter).ServeHTTP
        github.com/gravitational/teleport/lib/httplib/httplib.go:82 github.com/gravitational/teleport/lib/httplib.MakeTracingHandler.func1
        net/http/server.go:2109 net/http.HandlerFunc.ServeHTTP
        go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp@v0.36.1/handler.go:204 go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp.(*Handler).ServeHTTP
        net/http/server.go:2947 net/http.serverHandler.ServeHTTP
        net/http/server.go:1991 net/http.(*conn).serve
        runtime/asm_amd64.s:1594 runtime.goexit
User Message: AccessDenied: Access Denied
        status code: 403, request id: 92d8b8b3-0a8d-4153-8fa3-8b076db19f22, host id:

**Reply by webvictim** [reply]
That looks more like the IAM policy that's attached to your instance is lacking a relevant permission. Did anything change recently?

The policy you'd need is outlined here: https://goteleport.com/docs/deploy-a-cluster/helm-deployments/aws/#s3-iam-policy

**Reply by suchisur** [reply]
@PallaviPalisetti 
Any updates on this?
